wagtail/wagtailadmin/decorators.py

```diff
@@ -3,26 +3,33 @@
 from django.contrib.auth.views import redirect_to_login as auth_redirect_to_login
 from django.core.urlresolvers import reverse
 from django.utils.translation import ugettext as _
+from django.http import HttpResponseForbidden
 
 from wagtail.utils.compat import user_is_anonymous
 from wagtail.wagtailadmin import messages
 
 
-def redirect_to_login(request):
-    return auth_redirect_to_login(
-        request.get_full_path(), login_url=reverse('wagtailadmin_login'))
+def reject_request(request):
+    if request.is_ajax():
+        return HttpResponseForbidden()
```

Comment:
I personally prefer raising `PermissionDenied` instead, as I supposed it creates a better error page when in DEBUG mode.